tests/mtouch/LocalizationTests.cs

```diff
@@ -0,0 +1,112 @@
+using System;
+using NUnit.Framework;
+using System.Globalization;
+using System.Threading;
+using System.Reflection;
+using System.Resources;
+using System.Text;
+using System.Collections.Generic;
+
+namespace Xamarin.Tests
+{
+	[TestFixture]
+	public class LocalizationTests
+	{
+		[TestCase ("cs-CZ")]
+		[TestCase ("de-DE")]
+		[TestCase ("es-ES")]
+		[TestCase ("fr-FR")]
+		[TestCase ("it-IT")]
+		[TestCase ("ja-JP")]
+		[TestCase ("ko-KR")]
+		[TestCase ("nl")]
+		[TestCase ("pt-BR")]
+		[TestCase ("pt-PT")]
+		[TestCase ("ru-RU")]
+		[TestCase ("sv")]
+		[TestCase ("tr-TR")]
+		[TestCase ("zh-CN")]
+		[TestCase ("zh-TW")]
+		public void TestSpecificErrorCode (string culture)
+		{
+			var errorCode = "MT0015";
+			var originalUICulture = Thread.CurrentThread.CurrentUICulture;
+			var originalCulture = Thread.CurrentThread.CurrentCulture;
+
+			try {
+				var englishError = TranslateError ("en-US", errorCode);
+				var newCultureError = TranslateError (culture, errorCode);
+				Assert.AreNotEqual (englishError, newCultureError, $"\"{errorCode}\" is not translated in {culture}.");
+			} catch (NullReferenceException){
+				Assert.IsFalse (true, $"Error code \"{errorCode}\" was not found");
```

Comment:
This is slightly simpler:

```suggestion
				Assert.Fail ($"Error code \"{errorCode}\" was not found");
```